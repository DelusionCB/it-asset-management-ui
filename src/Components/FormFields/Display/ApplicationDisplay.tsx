import React from 'react';
import {Container, Row} from 'reactstrap';
import {
    DisplayArray,
    DisplayBoolean,
    DisplayDescription,
    DisplayHeader,
    DisplayObject, DisplayStatus,
    DisplayText, DisplayTextArray,
} from '../../CustomComponents/DisplayComponents';
import './index.scss';
import {appDisplayPropTypes} from '../../../Types/types.displays';
import ActionButtons from '../../ActionButtons/ActionButtons';

function ApplicationDisplay ({values, t, navigate}: appDisplayPropTypes): JSX.Element {
    return (
        <Container className='directory-wrapper'>

            <DisplayHeader t={t} label='application.app' />

            <DisplayDescription t={t} label='application.description' />

            <DisplayText t={t} value={values.name} label='application.values.name' />
            <DisplayText t={t} value={values.description} label='application.values.description' />
            <DisplayText t={t} value={values.classification} label='application.values.classification' />
            <DisplayTextArray t={t} value={values.customership} label='application.values.customership' />
            <DisplayStatus t={t} value={values.application_status} label='application.values.application_status' />
            <DisplayBoolean t={t} value={values.person_register} label='application.values.person_register' />
            <DisplayBoolean t={t} value={values.personal_info_logging} label='application.values.personal_info_logging' />
            <DisplayText t={t} value={values.install_info} label='application.values.install_info' />

            <DisplayText t={t} value={values.keywords} label='application.values.keywords' />

            <DisplayDescription t={t} label='application.dependency' />

            <DisplayArray t={t} navigate={navigate} value={values.installed_server} label='application.values.installed_server' />
            <DisplayArray navigate={navigate} t={t} value={values.service_dependency} label='application.values.service_dependency' />
            <DisplayArray navigate={navigate} t={t} value={values.integration} label='application.values.integration' />
            <DisplayObject t={t} navigate={navigate} value={values.license} label='application.values.license' />
            <DisplayObject t={t} navigate={navigate} value={values.contract} label='application.values.contract' />
            <DisplayArray t={t} navigate={navigate} value={values.application_dependency} label='application.values.application_dependency' />

            <DisplayDescription t={t} label='application.continuity' />

            <DisplayText t={t} value={values.update_practice} label='application.values.update_practice' />
            <DisplayText t={t} value={values.security_practice_monitoring} label='application.values.security_practice_monitoring' />
            <DisplayText t={t} value={values.recovery_practices_convalescence} label='application.values.recovery_practices_convalescence' />
            <DisplayText t={t} value={values.user_rights_management} label='application.values.user_rights_management' />
            <DisplayText t={t} value={values.security_solutions} label='application.values.security_solutions' />

            <DisplayDescription t={t} label='application.responsibility' />

            <DisplayText t={t} value={values.product_owner} label='application.values.product_owner' />
            <DisplayText t={t} value={values.application_holder} label='application.values.application_holder' />
            <DisplayText t={t} value={values.admin_users} label='application.values.admin_users' />
            <DisplayText t={t} value={values.liability_professional_users} label='application.values.liability_professional_users' />
            <DisplayText t={t} value={values.holder_extra_info} label='application.values.holder_extra_info' />

            <Row>
                <h3>Käyttöoikeusvastaavat:</h3>
                <p></p>
            </Row>

            <DisplayObject navigate={navigate} t={t} value={values.provider} label='application.values.provider' />
            <DisplayText t={t} value={values.provider_responsibility} label='application.values.provider_responsibility' />
            <DisplayText t={t} value={values.additional_contacts} label='application.values.additional_contacts' />

            <DisplayDescription t={t} label='application.extra' />

            <DisplayText t={t} value={values.known_issues} label='application.values.known_issues' />
            {/* <DisplayText t={t} value={values.fileUrl} label='values.fileUrl' /> */}
            <ActionButtons
                values={values}
                type={'application'}
                mode={'display'}
            />
        </Container>
    );
}

export default ApplicationDisplay;
