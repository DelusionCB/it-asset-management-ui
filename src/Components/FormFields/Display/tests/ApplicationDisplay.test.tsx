import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ApplicationDisplay from '../ApplicationDisplay';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {appDefaultValues} from '../../../../Types/types.directories';

describe('<ApplicationDisplay />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ApplicationDisplay t={jest.fn()} navigate={jest.fn()} values={appDefaultValues} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
