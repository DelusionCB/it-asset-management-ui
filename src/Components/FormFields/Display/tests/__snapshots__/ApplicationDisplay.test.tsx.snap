// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationDisplay /> Snapshot renders correctly 1`] = `
<div
  className="directory-wrapper container"
>
  <div
    className="display-header row"
  >
    <h1 />
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p>
        tosi
      </p>
    </div>
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <h4 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3>
        :
      </h3>
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3>
        :
      </h3>
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <h4 />
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <h4 />
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <h4 />
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <div>
      <h4 />
    </div>
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <div>
      <h4 />
    </div>
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <h4 />
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="row"
  >
    <h3>
      Käyttöoikeusvastaavat:
    </h3>
    <p />
  </div>
  <div
    className="display row"
  >
    <h3>
      :
    </h3>
    <div>
      <h4 />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="display-description row"
  >
    <h2 />
  </div>
  <div
    className="display row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="action-buttons"
  >
    <button
      className="action-button btn btn-secondary disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      button.copy
    </button>
    <button
      className="action-button btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      button.delete
    </button>
    <button
      className="action-button btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      button.edit
    </button>
  </div>
</div>
`;
